app/src/main/java/com/norvellium/tasky/agenda/presentation/agenda/AgendaFragment.kt

```diff
@@ -0,0 +1,53 @@
+package com.norvellium.tasky.agenda.presentation.agenda
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.norvellium.tasky.R
+import com.norvellium.tasky.databinding.FragmentAgendaBinding
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+
+@AndroidEntryPoint
+class AgendaFragment : Fragment(), CoroutineScope by MainScope() {
+
+    companion object {
+        fun newInstance() = AgendaFragment()
+    }
+
+    private val viewModel: AgendaViewModel by viewModels()
+
+    private var _binding: FragmentAgendaBinding? = null
+
+    // This property is only valid between onCreateView and onDestroyView.
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentAgendaBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // This button is just a placeholder to allow logout functionality
+        binding.btnLogout.setOnClickListener {
+            viewModel.logout()
+            findNavController().popBackStack()
+            findNavController().navigate(R.id.nav_login)
```

Comment:
Thanks, I've been looking for something like this, didnt realize there was a better way to do it but this will help alot. I'm slightly confused on how to use it though. The method signature you wrote doesnt seem to exist (navigate with int, navoptions). There is one that utilizes a int, bundle, but since I've not done this method before, I'm not sure what you had intended. Also what does `R.id.main_nav_graph` correspond to? Is this my `mobile_navigation.xml` file in my navigation resources directory? Or the fragment hosting the nav graph, in this case my main activity?

Reply:
Ah yeah, I always use it with the navigation safe args plugin, but wasn't sure if you use that. You can add it to your build.gradle project file and it will generate classes that allow you to pass args in a type-safe manner. Then the syntax will look similar to this:
```
findNavController().navigate(
    MainNavGraphDirections.globalActionToLoginFragment(),
    NavOptions.Builder()
        .setPopUpTo(R.id.main_nav_graph, true)
        .build()
)
```